core/domain/translation_domain.py

```diff
@@ -18,9 +18,262 @@
 
 from __future__ import annotations
 
+import enum
+
+from core import feconf
 from core import utils
 
-from typing import Dict
+from typing import Dict, List
+from typing_extensions import TypedDict
+
+
+class TranslatableContentFormat(enum.Enum):
+    """Represents all possible data types for any translatable content."""
+
+    HTML = 'html'
+    UNICODE_STRING = 'unicode'
+    SET_OF_NORMALIZED_STRING = 'set_of_normalized_string'
+    SET_OF_UNICODE_STRING = 'set_of_unicode_string'
+
+
+class TranslatableContentDict(TypedDict):
+    """Dictionary representing TranslatableContent object."""
+
+    content_id: str
+    content: feconf.ContentInTranslatableContent
```

Comment:
Let's perhaps call the type of this (in feconf) TranslatableContentValue. Otherwise there are too many things that sound like "content" floating around, it's hard for the reader to keep it straight in their head. (At least, it's hard for me :) )